### DISCUSSION TITLE
[G030Cx] Added support, questions about contribution!?

### DISCUSSION THREAD

**Discussion by hitech95** [stm32duino/Arduino_Core_STM32#1546] [Q&A] [answered]
I did some work on adding support for thoose chips.
So far is working fine (C8T variant).

https://github.com/hitech95/Arduino_Core_STM32/commits/feature/G030CxT
I don't undertand the thing about the **atomic** commits.
Should I `--sign-off` the work done?

Any suggestion before opening the PR?

**Comment by fpistm** [comment, marked as answer]
Hi @hitech95 
All is explained here but I guess this is the reason you ask this.
https://github.com/stm32duino/Arduino_Core_STM32/blob/main/CONTRIBUTING.md#4-commit-messages

Atomic means to split each specific dev into specific commit. 1 commit = 1 subject.
This allows to easily rework the PR or revert one thing if needed. If all the code is only in one commit it is hard to revert one dedicated part of the code.
This is stated here: https://www.freshconsulting.com/insights/blog/atomic-commits/
```
Benefits

    Easy to roll back without affecting other changes
    Easy to make other changes on the fly
    Easy to merge features to other branches
```
About signed-off:
`Sign-off is a requirement for getting patches into the Linux kernel and a few other projects, but most projects don't actually use it.`

It is not mandatory but it helps to find commit by signed-off for example.

**Comment by hitech95** [comment]
Another question, wich naming convention should I use?, i've seen somne convetional commits and some other more "random" description in the subject.

Most imporatant question is what to set the framework version in the Readme.md
I've updated my commit subjects to be capped and added the description in one.

**Comment by fpistm** [comment]
About naming convention, now I try to use conventional commit syntax to be able to use automatic changelog generation
https://www.conventionalcommits.org/en/v1.0.0/

In the Readme in general I set the next milestone in bold and 💛
